allennlp/common/file_utils.py

```diff
@@ -86,7 +86,12 @@ def __init__(
     def acquire(self, timeout=None, poll_interval=0.05):
         try:
             super().acquire(timeout=timeout, poll_intervall=poll_interval)
-        except (PermissionError, OSError):
+        except (PermissionError, OSError) as err:
+            if isinstance(err, OSError):
+                # OSError could be a lot of different things, but what we're looking
+                # for in particular is Linux errno 30 - EROFS - "Read-only file system".
```

Comment:
Well, this is interesting. When the whole file system is read-only, we're guaranteed to not have concurrent writers, so we don't even need to print the warning. I don't know if you want to get that granular though ...

Reply:
> When the whole file system is read-only, we're guaranteed to not have concurrent writers

That's not necessarily true. As @vikigenius mentioned above, this error can occur from within a Docker container while trying to access an attached volume. So while it is true that you couldn't have concurrent writers within the same Docker container, you could have concurrent writers on separate Docker containers or on the host OS itself.

Reply:
Ah, yes.